### PR TITLE
feat(cloudtrail): add configuration for IsOrganizationTrail

### LINKED ISSUE #21578: (cloudtrail): (configuration for IsOrganizationTrail is failed)
### Describe the feature

I miss the fact that I can use boolean to specify an OrganizationTrail as in CloudFormation.

https://docs.aws.amazon.com/de_de/AWSCloudFormation/latest/UserGuide/aws-resource-cloudtrail-trail.html#cfn-cloudtrail-trail-isorganizationtrail

### Use Case

I am currently converting my payer configuration to CDK, at least what is available at all, which unfortunately is still very little at the moment. Using CloudFormation, I can specify whether a trail is an organization trail and whether it is then automatically rolled out throughout the organization.
Unfortunately, this is not yet possible with CDK.

### Proposed Solution

Give an option that you can configure this using boolean, the default should be "false".

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.37.1

### Environment details (OS name and version, etc.)

none

### PR DESCRIPTION
**Pull request by daschaa**
Fixes #21578

Please add `pr-linter/exempt-readme` label since this property needs no entry in the README imho.

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts [hunk 1]
```diff
@@ -115,6 +115,17 @@ export interface TrailProps {
    * @default - if not supplied a bucket will be created with all the correct permisions
    */
   readonly bucket?: s3.IBucket;
+
+  /**
+   * Specifies whether the trail is applied to all accounts in an organization in AWS Organizations, or only for the current AWS account.
+   * The default is false, and cannot be true unless the call is made on behalf of an AWS account that is the management accountfor an organization in AWS Organizations.
+   * If the trail is not an organization trail and this is set to true, the trail will be created in all AWS accounts that belong to the organization.
+   * If the trail is an organization trail and this is set to false, the trail will remain in the current AWS account but be deleted from all member accounts in the organization.
+   *
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts:124]: This documentation feels a little clunky to me. After the first read through I felt like I didn't quite get it so I went and looked up `CfnTrail` to try to get clarification. I see that it came from there but I think we can make this more clear.

File: packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts [hunk 2]
```diff
@@ -285,6 +296,7 @@ export class Trail extends Resource {
       cloudWatchLogsRoleArn: logsRole?.roleArn,
       snsTopicName: this.topic?.topicName,
       eventSelectors: this.eventSelectors,
+      isOrganizationTrail: props.isOrganizationTrail ?? false,
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts:299]: What's the behavior here if the account isn't the owner of an organization and this gets set to true? Does it just fail silently? If so, do we have a way to validate that the account is actually the owner of an organization?

Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-cloudtrail/lib/cloudtrail.ts:299]:
```suggestion
      isOrganizationTrail: props.isOrganizationTrail,
```
Let's just pass in what's provided. That way it doesn't impact templates unless it's explicitly provided.

**Review by TheRealAmazonKendra — CHANGES_REQUESTED** [review]
I think I would actually like a README update on this because the CloudFormation documentation isn't clear enough. I would have been inclined to agree with you if it hadn't taken me multiple times reading it to understand what the actual behavior was.

**Comment by daschaa** [comment]
@TheRealAmazonKendra I have added an entry in the README, updated the docstring and implemented your suggestion to not use the `?? false` operator. Thanks for your valuable feedback!
Regarding the check if an account manages an organization, i did not find anything that could solve this problem. Do you know any way to check this?

**Comment by TheRealAmazonKendra** [comment]
> @TheRealAmazonKendra I have added an entry in the README, updated the docstring and implemented your suggestion to not use the `?? false` operator. Thanks for your valuable feedback! Regarding the check if an account manages an organization, i did not find anything that could solve this problem. Do you know any way to check this?

I'm not sure that there is a good way without introducing a LOT of extra complexity to this.

**Review by TheRealAmazonKendra — APPROVED** [review]

**Pull request merged**